doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -926,14 +909,257 @@ the conversion in hypotheses :n:`{+ @ident}`.
       proof term. See `#12200
       <https://github.com/coq/coq/issues/12200>`_ for more details.
 
-Conversion tactics applied to hypotheses
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _controlling-the-reduction-strategies:
+
+Controlling the reduction strategies and the conversion algorithm
+----------------------------------------------------------------------
+
+Coq provides several reduction strategies that tactics can invoke and three
+different algorithms to check the convertibility of types:
+
+- :tacn:`lazy` lazily compares applicative terms.
+- :tacn:`vm_compute` is a brute-force but efficient algorithm that
+  first normalizes the
+  terms before comparing them. It is based on a
+  bytecode representation of terms similar to the bytecode
+  representation used in the ZINC virtual machine :cite:`Leroy90`. It is
+  especially useful for intensive computation of algebraic values, such
+  as numbers, and for reflection-based tactics.
+- :tacn:`native_compute` uses the same algorithm as :tacn:`vm_compute`,
+  but it's based on converting the Coq code to OCaml.
+
+The commands to fine-tune the reduction strategies and the lazy conversion
+algorithm are described first.
+
+.. cmd:: Opaque {+ @reference }
+
+   Marks the specified constants so tactics won't unfold them with δ-conversion.
+   :gdef:`Unfolding <unfolding>` means to replace a constant by its definition.
+   "Constants" are items defined by :cmd:`Definition` or :cmd:`Let` (with an
+   explicit body), or by a command associated with a definition such as
+   :cmd:`Fixpoint`, etc, or by a proof ended by :cmd:`Defined`.
+
+   .. todo PR: can we be more explicit about the `Fixpoint`, etc.
```

Comment:
Maybe link the "Top-level definitions" and "Assertions and proofs" and "Top-level recursive functions" and "Advanced recursive functions" sections which talk about `Definition`, `Example, `@thm_token`, `Fixpoint`, `CoFixpoint`, and `Function`.  But note also that `Derive` (from the derive plugin in the stdlib), and `Instance` can also define constants, and many plugins (such as Equations) define other ways to add constants.  Furthermore, constants may be introduced as a side-effect of `Inductive` and `Record` (namely, the elimination schemes, and the equality schemes if `Decidable Equality Schemes` and/or `Boolean Equality Schemes` are set), by `Scheme`, by the `transparent_abstract` tactic, and even by `rewrite` (if the relevant rewriting scheme is not already present in the environment).  So, uh, no, it's not really possible to give an exhaustive list, especially in the presence of plugins that extend the vernacular.

Reply:
Note also that application of module functors creates constants, as does `Include`ing a module.